src/foam/classloader/WebModelFileDAO.js

```diff
@@ -52,10 +52,19 @@ foam.CLASS({
         var model;
         var foamCLASS = foam.CLASS;
 
-        foam.CLASS = function(m) {
-          var cls = m.class ? foam.lookup(m.class) : foam.core.Model;
-          model = cls.create(m, self);
-          foam.CLASS = foamCLASS;
+        foam.CLASS = function(clsDef) {
+          var classOfModel = clsDef.class ? foam.lookup(clsDef.class) :
```

Comment:
You've convinced me. PTAL at comments added. The only "rename" change I kept is not double-declaring `cls`. I think using the same variable to mean something entirely different was what instigated the confusion for me.